### DISCUSSION TITLE
how to import commonjs module when use 'import'

### DISCUSSION THREAD

**Discussion by rikisamurai** [pmndrs/jotai#1985] [Q&A]
in jotai package.json, when write 'import xx from 'jotai' ', node applys esm default
```json
  "exports": {
    ".": {
      "types": "./index.d.ts",
      "import": {
        "types": "./esm/index.d.mts",
        "default": "./esm/index.mjs"
      },
      "default": "./index.js"
    },
  },
```
but some browsers haven't support esm or es6, it will cause crash.
So i want to import commonjs when i write "`import { atom, useAtom } from 'jotai'` "

**Comment by Aslemammad** [comment]
Browsers not supporting ESM would not be able to support cjs which is a nodejs way of handling modules AFAIK. So this is a legacy transpiling issue which can be solved using the [vite plugin legacy](https://github.com/vitejs/vite/tree/main/packages/plugin-legacy). 


The community is moving towards ESM right now to make it the default way. So this is not related to Jotai.

**Reply by Aslemammad** [reply]
Feel free to reopen.
